### PR TITLE
CASSANDRA-18242 trunk

### PR DESCRIPTION
**Pull request by jonmeredith**
See https://issues.apache.org/jira/browse/CASSANDRA-18242

### CODE REVIEW

File: ide/idea/workspace.xml [hunk 1]
```diff
@@ -173,7 +183,28 @@
...
+      <option name="VM_PARAMETERS" value="
+                                          -Dcassandra.config=file://$PROJECT_DIR$/test/conf/cassandra.yaml
+                                          -Dcassandra.logdir=$PROJECT_DIR$/build/test/logs
+                                          -Dcassandra.reads.thresholds.coordinator.defensive_checks_enabled=true
+                                          -Dcassandra.ring_delay_ms=1000
```
Comment (dcapwell) [inline ide/idea/workspace.xml:189]: thanks for dedupping =)

File: ide/idea/workspace.xml [hunk 2]
```diff
@@ -173,7 +183,28 @@
...
+                                          -Dcassandra.skip_sync=true
```
Comment (dcapwell) [inline ide/idea/workspace.xml:190]: thanks for dedupping =)

File: ide/idea/workspace.xml [hunk 3]
```diff
@@ -173,7 +183,28 @@
...
+                                          -Dcassandra.strict.runtime.checks=true
+                                          -Dcassandra.test.flush_local_schema_changes=false
+                                          -Dcassandra.test.messagingService.nonGracefulShutdown=true
+                                          -Dcassandra.tolerate_sstable_size=true
+                                          -Dcassandra.use_nix_recursive_delete=true
+                                          -Dinvalid-legacy-sstable-root=$PROJECT_DIR$/test/data/invalid-legacy-sstables
```
Comment (dcapwell) [inline ide/idea/workspace.xml:196]: thanks for dedupping =)

File: ide/idea/workspace.xml [hunk 4]
```diff
@@ -173,7 +183,28 @@
...
+                                          -Djava.library.path=$PROJECT_DIR$/lib/sigar-bin
+                                          -Dlegacy-sstable-root=$PROJECT_DIR$/test/data/legacy-sstables
```
Comment (dcapwell) [inline ide/idea/workspace.xml:198]: thanks for dedupping =)

File: ide/idea/workspace.xml [hunk 5]
```diff
@@ -173,7 +183,28 @@
...
+                                          -Dlogback.configurationFile=file://$PROJECT_DIR$/test/conf/logback-test.xml
+                                          -Dmigration-sstable-root=$PROJECT_DIR$/test/data/migration-sstables
+                                          -XX:ActiveProcessorCount=2
+                                          -XX:HeapDumpPath=build/test
+                                          -XX:MaxMetaspaceSize=1G
+                                          -XX:SoftRefLRUPolicyMSPerMB=0
+                                          -ea
+                                          -Didea.click.expand.for.rest.of.options" />
```
Comment (dcapwell) [inline ide/idea/workspace.xml]: this one is new... quick google and didn't find results, what is this?

File: ide/idea/workspace.xml [hunk 6]
```diff
@@ -143,7 +143,17 @@
...
+                                          -DQT_SHRINKS=0
+                                          -Dcassandra.config=file://$PROJECT_DIR$/conf/cassandra.yaml
+                                          -Dcassandra.logdir=$PROJECT_DIR$/data/logs
+                                          -Dcassandra.reads.thresholds.coordinator.defensive_checks_enabled=true
+                                          -Dcassandra.storagedir=$PROJECT_DIR$/data
+                                          -Djava.library.path=$PROJECT_DIR$/lib/sigar-bin
+                                          -Dlogback.configurationFile=file://$PROJECT_DIR$/conf/logback.xml
+                                          -ea
+                                          -XX:HeapDumpPath=build/test
+                                          -Didea.click.expand.for.rest.of.options" />
```
Comment (dcapwell) [inline ide/idea/workspace.xml]: this is new

File: ide/idea/workspace.xml [hunk 7]
```diff
@@ -193,7 +224,18 @@
...
+                                          -Dcassandra-foreground=yes
+                                          -Dcassandra.config=file://$PROJECT_DIR$/conf/cassandra.yaml
+                                          -Dcassandra.jmx.local.port=7199
+                                          -Dcassandra.logdir=$PROJECT_DIR$/data/logs
+                                          -Dcassandra.reads.thresholds.coordinator.defensive_checks_enabled=true
+                                          -Dcassandra.storagedir=$PROJECT_DIR$/data
+                                          -Djava.library.path=$PROJECT_DIR$/lib/sigar-bin
+                                          -Dlogback.configurationFile=file://$PROJECT_DIR$/conf/logback.xml
+                                          -Xmx1G
+                                          -ea
```
Comment (dcapwell) [inline ide/idea/workspace.xml]: why did you remove `-XX:HeapDumpPath=build/test`?

**Review by dcapwell — CHANGES_REQUESTED** [review]
only blocking comment is the lack of `-XX:HeapDumpPath=build/test` in the last VM param; once that is fixed I am +1

File: ide/idea/workspace.xml (hunk 7 above)
Comment (jonmeredith) [reply ide/idea/workspace.xml]: excellent eyes - mistake, will add back in.

File: ide/idea/workspace.xml (hunk 5 above)
Comment (jonmeredith) [reply ide/idea/workspace.xml]: It was an attempt to remind somebody looking at the IDEA GUI that there were options that weren't visible due to all the whitespace. As it caused confusion in review I'll just remove it.

**Review by jonmeredith — COMMENTED** [review]
Thanks for the review
